integration/kafka/src/main/java/alluxio/kafka/connect/AlluxioSinkTask.java

```diff
@@ -0,0 +1,177 @@
+      if (!mFs.exists(tmpTopicsDirPath)) {
+        mFs.createDirectory(tmpTopicsDirPath);
+      }
+    } catch (IOException | AlluxioException e) {
+      LOG.error("Exception is {}", e);
+      throw new RuntimeException("Failed to create " + topDir + " dir");
+    }
+
+    Collection<TopicPartition> assignments = context.assignment();
+    try {
+      initializeTopicPartitionWriter(assignments);
+    } catch (IOException | AlluxioException e) {
+      LOG.error("Exception is {}", e);
+      throw new RuntimeException("Failed to initialize TopicPartitionWriter");
+    }
+  }
+
+  /**
+   * Initialize AlluxioTopicPartitionWriter according to task assignments.
+   * One TopicPartition corresponds to one stream.
+   *
+   * @param assignments assigned TopicPartitions for this task
+   */
+  private void initializeTopicPartitionWriter(Collection<TopicPartition> assignments)
+    throws AlluxioException, IOException {
+    for (TopicPartition tp : assignments) {
+      AlluxioTopicPartitionWriter tpWriter =
+          new AlluxioTopicPartitionWriter(mConfig, context, mFormat, mFs, tp);
+      tpWriter.initialize();
+      mTpWriters.put(tp, tpWriter);
+    }
+  }
+
+  @Override
+  public void put(Collection<SinkRecord> records) {
+    for (SinkRecord record : records) {
+      String topic = record.topic();
+      int partition = record.kafkaPartition();
+      TopicPartition topicPartition = new TopicPartition(topic, partition);
+      AlluxioTopicPartitionWriter writer = mTpWriters.get(topicPartition);
+      if (writer != null) {
+        writer.writeToBuffer(record);
+      } else {
+        LOG.error("Failed to get writer of " + topicPartition.toString());
+      }
+    }
+
+    for (AlluxioTopicPartitionWriter writer : mTpWriters.values()) {
+      try {
+        writer.writeRecord();
+      } catch (IOException | AlluxioException e) {
+        LOG.error("Failed to write record to " + writer.getTopicPartition().toString());
+        LOG.error("Exception is {}", e);
+      }
+    }
+  }
+
+  /**
+   * Flushes all records that have been put for the specified topic-partitions. The
+   * offsets are provided for convenience, but could also be determined by tracking all offsets
+   * included in the SinkRecords passed to put method.
+   *
+   * @param offsets mapping of TopicPartition to committed offset
+   */
+  @Override
+  public void flush(Map<TopicPartition, OffsetAndMetadata> offsets) {
+  }
+
+  /**
+   * The SinkTask uses this method to create writers for newly assigned partitions in case of
+   * partition rebalance.
+   *
+   * @param partitions The list of partitions that are now assigned to the task (may include
+   *                   partitions previously assigned to the task)
+   */
+  @Override
+  public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+    //TODO(GuangHui): Create writers for newly assigned partitions in case of partition rebalance
```

Comment:
These are really important to achieve any delivery guarantee. It will be really nice to add them before merging to master. @zhutongxue Let me know if you want to discuss more.